### ISSUE TITLE
package manager fails to build

### ISSUE THREAD

**Issue by lancedolan** [ist-dresden/composum#60]
Using Sling 8. I can provide a full list of all my OSGI bundles installed, if you'd like.

- I created a package and gave it a filter of /content. Underneath /content, I have several websites worth of node data, but nothing overwhelming.
- When I clicked "(Re)build this package" it would only include content at a particular path (/content/edliohs/home/jcr:content/spotlight/message1/Principal Message 8.23.16.pdf) and wouldn't include any of the other hundreds of nodes throughout the content tree. It behaved as if I had an exclude rule excluding everything under /content except that node.
- I deleted that node.
- It will now include nothing under /content
- this is the output

A  META-INF
  A  META-INF/vault
  A  META-INF/vault/config.xml
  A  META-INF/vault/filter.xml
  A  META-INF/vault/nodetypes.cnd
  A  META-INF/vault/properties.xml
  A  /.content.xml
  A  /content
  A  /content/.content.xml
  -  Aggregation status: 1 of 0 prepared, 0 collected
  A  META-INF/vault/definition/.content.xml

this is the contents of the filter

```
<?xml version="1.0" encoding="UTF-8"?>
<workspaceFilter version="1.0">
    <filter root="/content"/>
</workspaceFilter>
```

There are no error or info messages in error.log

I am lost.

**Comment by lancedolan** [comment]
Note: it can and will build anything else, for example the /apps directory. Just nothing under content.

**Comment by ist-rw** [comment]
The package manger is a user interface for the Jackrabbit Vault framework. The rules to build a package are implemented in this framework. But for single files a package the simplest way to specify the filter is a list of filters with the complete path to the file as the filters root path. I've attached an example for one file.
[issue-60-content-1.0.zip](https://github.com/ist-dresden/composum/files/757139/issue-60-content-1.0.zip)

**Comment by lancedolan** [comment]
Yes, all of this I know. I'm not new to JCR or package manager concept (I used CRX for several years). In my issue description, you can see that my filter root path is /content. It's behaving as if there is an exclude rule or include rule that I didn't include, such that nothing will be included in my content package except this one single path (/content/edliohs/home/jcr:content/spotlight/message1/Principal Message 8.23.16.pdf). 

In other words, I build a /content package, and it's empty when I download it.

I'm wondering if there is some sort of cache of filter rules that composum keeps in the JCR somewhere? The list of reasons this could be happening is long, but composum is definitely on that list in my assessment.

**Comment by lancedolan** [comment]
The "output" that i provided above is the output from Composum Package Manager when it builds a package that I've just created with only one filter (/content) and there are hundreds of nodes underneath /content.

**Comment by stoerr** [comment]
Thank you for the report! This does not look like it is a Composum package manager issue at all, since it just delegates this operation to the Jackrabbit implementation. So there would not be much we could do.
Since this is pretty old, anyway, and likely not reproducible, I'll close this issue.
